contracts/EventBasedPredictionMarket.sol

```diff
@@ -0,0 +1,318 @@
+     * @param shortTokensToRedeem number of short tokens to settle.
+     * @return collateralReturned total collateral returned in exchange for the pair of synthetics.
+     */
+    function settle(uint256 longTokensToRedeem, uint256 shortTokensToRedeem)
+        public
+        returns (uint256 collateralReturned)
+    {
+        // Get the settlement price and store it. Reverts if price has not yet been resolved.
+        if (!receivedSettlementPrice) getExpirationPrice();
+
+        require(longToken.burnFrom(msg.sender, longTokensToRedeem));
+        require(shortToken.burnFrom(msg.sender, shortTokensToRedeem));
+
+        // expiryPercentLong is a number between 0 and 1e18. 0 means all collateral goes to short tokens and 1e18 means
+        // all collateral goes to the long token. Total collateral returned is the sum of payouts.
+        uint256 longCollateralRedeemed = (longTokensToRedeem * collateralPerPair * expiryPercentLong) / (1e18**2);
+        uint256 shortCollateralRedeemed = (shortTokensToRedeem * collateralPerPair * (1e18 - expiryPercentLong)) /
+            (1e18**2);
+
+        collateralReturned = longCollateralRedeemed + shortCollateralRedeemed;
+        collateralToken.safeTransfer(msg.sender, collateralReturned);
+
+        emit PositionSettled(msg.sender, collateralReturned, longTokensToRedeem, shortTokensToRedeem);
+    }
+
+    /****************************************
+     *          INTERNAL FUNCTIONS          *
+     ****************************************/
+
+    /**
+     * @notice Request a price in the optimistic oracle for a given request timestamp and ancillary data combo. Set the bonds
+     * accordingly to the deployer's parameters. Will revert if re-requesting for a previously requested combo.
+     */
+    function _requestOraclePrice() internal {
+        OptimisticOracleInterface optimisticOracle = getOptimisticOracle();
+
+        collateralToken.safeApprove(address(optimisticOracle), proposerReward);
+
+        optimisticOracle.requestPrice(
+            priceIdentifier,
+            expirationTimestamp,
+            customAncillaryData,
+            collateralToken,
+            proposerReward
+        );
+
+        // Set the Optimistic oracle liveness for the price request.
+        optimisticOracle.setCustomLiveness(
+            priceIdentifier,
+            expirationTimestamp,
+            customAncillaryData,
+            optimisticOracleLivenessTime
+        );
+
+        // Set the Optimistic oracle proposer bond for the price request.
+        optimisticOracle.setBond(
+            priceIdentifier,
+            expirationTimestamp,
+            customAncillaryData,
+            optimisticOracleProposerBond
+        );
+
+        // Make the request an event-based request.
+        OptimisticOracleInterfaceEventBased(address(optimisticOracle)).setEventBased(
+            priceIdentifier,
+            expirationTimestamp,
+            customAncillaryData
+        );
```

Comment:
Since the plan is to use the new OO, I believe you'll also need to call `setCallbacks` here to use the callback in `priceDisputed`